localstack/utils/analytics/event_publisher.py

```diff
@@ -84,7 +84,7 @@ def read_api_key_safe():
 
 
 def _get_config_file(path):
-    get_or_create_file(path)
+    get_or_create_file(path, permissions=0o600)
```

Comment:
I did not use the new save_config_file method in here, because I am not sure whether this whole logic is still necessary? There is an TODO about it there as well.

Reply:
👍 Good catch - looks like we can simplify/streamline this part a bit, to address the `TODO`. We could tackle this as a follow-up PR when we get a chance (not super urgent, though).